### PR TITLE
update .travis.yml with latest recommendation from atom/ci

### PR DESCRIPTION
**Pull request by dirk-thomas**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,4 +1,35 @@
...
+  depth: 10
+
+sudo: false
+
+os:
+  - linux
+  - osx
+
+env:
+  global:
```
Comment (Arcanemagus) [inline .travis.yml]: This section isn't actually necessary unless you are specifying test packages (which `atom-package-deps` should be handling for you :wink:).

Comment (dirk-thomas) [inline .travis.yml]: Thank you for the hint. I removed the `global` entry.

**Comment by dirk-thomas** [comment]
@Arcanemagus I can't reproduce these failures locally. I am inclined to close the PR since the change isn't necessary. Any idea what could be different - I already tried various configurations?

**Comment by Arcanemagus** [comment]
That is just bizarre, that's almost identical to what several other projects are using with no issues. Seeing if I can spot anything.

**Comment by Arcanemagus** [comment]
Something is wonky with it... it's running scripts on the wrong workers (before it was running the Linux scripts on an OSX worker, now the Linux is on the proper worker, but the OS X looks to be odd.

**Comment by dirk-thomas** [comment]
I fixed the line length warning - that was obvious. But all the marker checks should be good as they are and shouldn't fail.

**Comment by Arcanemagus** [comment]
If you check the output it seems something is off in the config, something about "invalid prexis 'xml'"

**Comment by Arcanemagus** [comment]
(Also, I guess the OS X workers just use Debian names in their `Build system information` :confused:, see [here](https://travis-ci.org/AtomLinter/linter-stylelint/jobs/123269612) and [here](https://travis-ci.org/AtomLinter/linter-xmllint/jobs/123394718))

**Comment by Arcanemagus** [comment]
I wonder if the version of `xmllint` on the Linux box is different?

**Comment by dirk-thomas** [comment]
My local machine is Linux. And neither on Trusty nor on Xenial I get the same result - it passes on both. And even the recent PR to update the AtomLinter version passed.

**Comment by Arcanemagus** [comment]
It's working locally for me too, so something has to be different in the CI environment, we just need to get enough debugging info to be reported to figure out what :wink:.

**Comment by Arcanemagus** [comment]
In the CI environment:
```raw
xmllint: using libxml version 20708
   compiled with: Threads Tree Output Push Reader Patterns Writer SAXv1 FTP HTTP DTDValid HTML Legacy C14N Catalog XPath XPointer XInclude Iconv ISO8859X Unicode Regexps Automata Expr Schemas Schematron Modules Debug Zlib 
```

Locally:
```raw
xmllint: using libxml version 20902
   compiled with: Threads Tree Output Push Reader Patterns Writer SAXv1 FTP HTTP DTDValid HTML Legacy C14N Catalog XPath XPointer XInclude Iconv ISO8859X Unicode Regexps Automata Expr Schemas Schematron Modules Debug Zlib Lzma
```

**Comment by dirk-thomas** [comment]
The CI might run on Ubuntu Precise again then? That will likely be too old to support the features this plugin uses. I can check how to ensure that Travis uses Trusty.

**Comment by Arcanemagus** [comment]
Yea, definitely running on Precise: `Build dist: precise`.

**Comment by Arcanemagus** [comment]
Yay for squashed commits to hide all these merge ones that keep popping up :confused:

**Comment by Arcanemagus** [comment]
There, one instance for each pair of linux/osx + stable/beta

**Comment by Arcanemagus** [comment]
Wow, the Trusty env doesn't have `node` installed.

**Comment by dirk-thomas** [comment]
It might be called `nodejs`...

**Comment by Arcanemagus** [comment]
[This](https://docs.travis-ci.com/user/languages/javascript-with-nodejs) says `node_js` at least.

**Comment by Arcanemagus** [comment]
Looks like a network failure in the linux stable build, restarting.

**Comment by Arcanemagus** [comment]
Cool, passing build!

It looks like Travis is already running `apt-get -yq update` in an earlier step, so let's remove that from the script.

**Comment by Arcanemagus** [comment]
Actually, I just remembered the OS X builds should have `node_js` support as well, might as well move that out of the OS X specific part.

**Comment by Arcanemagus** [comment]
Silly Travis... that created a 5th build for some reason :confused:.

**Comment by Arcanemagus** [comment]
Any objection to removing the beta build on OS X? Travis-CI has a rather limited supply of public OSX containers and they cause large queue times.

**Comment by dirk-thomas** [comment]
Fine by me.

**Comment by Arcanemagus** [comment]
Done :wink:

**Comment by Arcanemagus** [comment]
Any other changes you want to stick in here? Did you want me to merge?

**Comment by dirk-thomas** [comment]
Thanks for all your effort on this. I am currently on a trip but I can take care of it as soon as I am back.

**Comment by Arcanemagus** [comment]
I'll go ahead and merge it, any further changes can just be in a new PR :wink:.

**Pull request merged**